### ISSUE TITLE
Switch BrowserModule Dependency to Common Module

### ISSUE THREAD

**Issue by lmalcom** [rsaenen/ngx-input-file#7]
Hey, 

I love your component but I had a problem integrating it into my application when using the angular router. This component imports the Browser Module component which breaks on Angular if something else imports it. I tried it out with the Common module and it works perfectly fine. Could you switch to using the Common Module instead so there is easier integration? 

Thanks!

**Comment by rsaenen** [comment]
Hi, thanks a lot !
I'm also using this component in my pet project (which works) with these imports:
```typescript
@NgModule({
    imports: [
        BrowserAnimationsModule,
        BrowserModule,
        FormsModule,
        HttpClientModule,
        InputFileModule,
        MyRoutingModule,
        ReactiveFormsModule
    ],
    ...
)}
```

```typescript
@NgModule({
    imports: [ RouterModule.forRoot(routes, { useHash: true }) ],
    exports: [ RouterModule ]
})
```

What is your error?

**Comment by lmalcom** [comment]
Hey, thanks for the fast response! 

My project is basically the same in this, the main problem being that I also import the `BrowserModule` in my parent `App.module` class. The input file module itself imports this in the source code and because of that I get this error when using it in my project. 

```
Error: BrowserModule has already been loaded. If you need access to common directives such as NgIf and NgFor from a lazy loaded module, import CommonModule instead.
Error: BrowserModule has already been loaded. If you need access to common directives such as NgIf and NgFor from a lazy loaded module, import CommonModule instead.
```

Simply changing `BrowserModule` to `CommonModule` in your base module fixes this. 

```
InputFileModule.decorators = [
    { type: NgModule, args: [{
                declarations: [
                    DropZoneDirective,
                    InputFileComponent
                ],
                imports: [
                    CommonModule,
                    FormsModule,
                    HttpClientModule
                ],
                exports: [
                    DropZoneDirective,
                    InputFileComponent
                ],
                providers: [],
                entryComponents: [InputFileComponent]
            },] },
];
```

I realize you might have exported it from your project, so maybe the Angular CLI automatically attaches the BrowserModule in your build code, so I think you just need to explicitly import it in your module's base code. 

Let me know if that works/makes sense or if I should do something else.

**Comment by rsaenen** [comment]
Do you use any lazy loading ?
Can you paste `ng -v` output ?

I guess switching to the CommonModule is not the right answer, the demo of this package import InputFileModule and BrowserModule and I don't reproduce this issue.
Moreover, this package has been released for 1 year, a simular issue was not opened so I'm thinking that the issue comes from another part.

Can you show me your app.module ?

**Comment by lmalcom** [comment]
Do you use any lazy loading ?
Yes

Can you paste ng -v output ?

```
Angular CLI: 1.7.1
Node: 8.9.4
OS: linux x64
Angular: 5.2.6
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router

@angular/cli: 1.7.1
@angular-devkit/build-optimizer: 0.3.2
@angular-devkit/core: 0.3.2
@angular-devkit/schematics: 0.3.2
@ngtools/json-schema: 1.2.0
@ngtools/webpack: 1.10.1
@schematics/angular: 0.3.2
@schematics/package-update: 0.3.2
typescript: 2.5.3
webpack: 3.11.0
```

Can you show me your app.module ?

```
/**
 * @license
 * Copyright Akveo. All Rights Reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 */
import { APP_BASE_HREF } from '@angular/common';
import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { NgModule } from '@angular/core';
import { HttpClientModule } from '@angular/common/http';
import { CoreModule } from './@core/core.module';

import { AppComponent } from './app.component';
import { AppRoutingModule } from './app-routing.module';
import { ThemeModule } from './@theme/theme.module';
import { NgbModule } from '@ng-bootstrap/ng-bootstrap';

@NgModule({
  declarations: [AppComponent],
  imports: [
    BrowserModule,
    BrowserAnimationsModule,
    HttpClientModule,
    AppRoutingModule,
    NgbModule.forRoot(),
    ThemeModule.forRoot(),
    CoreModule.forRoot(),
  ],
  bootstrap: [AppComponent],
  providers: [
    { provide: APP_BASE_HREF, useValue: '/' },
  ],
})
export class AppModule {
}
```

Maybe you're right and I can find a way of working with it if lazy loading is enabled and if so then awesome! Let me know if this helps.

**Comment by rsaenen** [comment]
Can you replace BrowserModule by BrowserAnimationModule? As BrowserAnimationModule import BrowserModule.
It seems to be an issue with lazy loading.
[#5684](https://github.com/angular/angular-cli/issues/5684)

**Comment by lmalcom** [comment]
I still get the error. The error doesn't get thrown when the application launches, it only gets thrown when I load the page that uses this module. 

Having them both on the base app works and also removing the BrowserModule also works, but only for the pages that don't use this module. 

If you want to check out the repo that I forked this from it's here, https://github.com/akveo/ngx-admin. If you wanted to recreate my flow you would have to create another module that uses this component and attach it to the router to see the problem with lazy loading.

**Comment by rsaenen** [comment]
It's a weird behavior, my bad... 
I do not have any objections to switch to the CommonModule if there's no regression.

Can you make a PR?

**Comment by lmalcom** [comment]
Cool I'll make one later today!

**Comment by michaeljota** [comment]
Hi. If this module should be imported in other Angular applications, then this should not import BrowserModule. This will break any actual app that tries to use it, because you can only import BrowserModule once in your application. This is not about lazy loading, although you may find yourself thinking about importing BrowserModule in your Features Modules, that won't work either you early or lazy load them.

**Comment by lmalcom** [comment]
Just created a pull request now. Sorry for the super long delay! https://github.com/rsaenen/ngx-input-file/pull/8

**Comment by rsaenen** [comment]
Thanks for contributing, I'll publish the new release asap.

**Comment by rsaenen** [comment]
Released in 1.0.3
